block/provider/provider.go

```diff
@@ -226,6 +226,21 @@ func (vpcp *VPCBlockProvider) OpenSession(ctx context.Context, contextCredential
 		maxRetryGap = vpcp.Config.VPCConfig.MaxRetryGap
 	}
 
+	//Update retry logic for custom retry with default values
+	if vpcp.Config.VPCConfig.MinVPCRetryGap > ConstMinVPCRetryGap && vpcp.Config.VPCConfig.MinVPCRetryGap < ConstantRetryGap {
+		minVPCRetryGap = vpcp.Config.VPCConfig.MinVPCRetryGap
+	}
+
+	if vpcp.Config.VPCConfig.MinVPCRetryGapAttempt > 0 {
+		minVPCRetryGapAttempt = vpcp.Config.VPCConfig.MinVPCRetryGapAttempt
+	}
+
+	if vpcp.Config.VPCConfig.MaxVPCRetryAttempt > ConstMaxVPCRetryAttempt {
+		maxVPCRetryAttempt = vpcp.Config.VPCConfig.MaxVPCRetryAttempt
```

Comment:
maxVPCRetryAttempt -> This should be dynamically calculated based on minVPCRetryGapAttempt, minVPCRetryGap each time.